contracts/Proofs.sol

```diff
@@ -50,17 +64,25 @@ contract Proofs {
     return periodOf(block.timestamp);
   }
 
+  /// @notice Informs the contract that proofs should be expected for id
+  /// @dev Requires that the id is not already in use
+  /// @param id identifies the proof expectation, typically a slot id
+  /// @param endId Identifies the id of the proof expectation ending. Typically a request id. Different from id because the proof ending is shared amongst many ids.
+  /// @param probability The probability that a proof should be expected
+  /// @param duration Duration, from now, for which proofs should be expected
   function _expectProofs(
-    bytes32 id,
+    bytes32 id, // typically slot id
+    bytes32 endId, // typically request id, used so that the ending is global for all slots
     uint256 probability,
     uint256 duration
   ) internal {
     require(!ids[id], "Proof id already in use");
     ids[id] = true;
     starts[id] = block.timestamp;
-    ends[id] = block.timestamp + duration;
+    ends[endId] = block.timestamp + duration;
```

Comment:
I wonder whether we shouldn't set the end in a separate function, that we can call from the marketplace once a request for storage is submitted. Then we only have to set it once.